scripts/hab_conf_transfer_to_Q.Rmd

```diff
@@ -0,0 +1,129 @@
+```
+
+
+## Purpose
+
+This workflow is unique to the 2023 fiss form data and only needs to be done once. While populating the `habitat_confirmations.xls` spreadsheet, edits were made by hand to the `temperature`, `conductivity`, and `turbidity` columns in the `step_2_fish_coll_data` sheet. While doing so we realized we should QA `form_fiss_2023` in Q before populating the `habitat_confirmations.xls` spreadsheet so that we shouldn't need to make edits by hand, like in this case. 
+
+But we learn from our mistakes and will now go and QA `form_fiss_2023`. We don't need to duplicate our edits, so we will copy the edits made by hand in `habitat_confirmations.xls` to `form_fiss_2023.gpkg` before starting the QA. The purpose of this script is to add data from the `habitat_confirmations.xls` spreadsheet to the `form_fiss_2023.gpkg`. `eval = FALSE` because we only need to run this code once, and don't want to run this chunk when knitting the document.
+
+Moving forward we should use the following workflow:
+
+  1. QA `form_fiss_2023` in Q.
+  2. Run `fiss_site_tidy.R` to populate step 1 and step 4 of the `habitat_confirmations.xls` spreadsheet.
+  3. Run `extract_inputs.R` to populate step 2 of the `habitat_confirmations.xls` spreadsheet.
+  4. Run `fiss_data_tidy.R` to populate step 3 of the `habitat_confirmations.xls` spreadsheet.
+
+
+# Define Variables and Import Data
+
+-  Mergin project directory
+-  Input and output files for this script
+-  Import the raw `form_fiss_2023.gpkg`
+-  Import the `habitat_confirmations.xls`
+
+```{r def-var, echo = TRUE}
+
+dir_project <- 'sern_skeena_2023'
+file_in <- 'scripts/hab_conf_transfer_to_Q.Rmd'
+# This assumes we use the same directory for the output file as the input file
+file_out <- 'hab_conf_transfer_to_Q.html'
+
+## Import the raw form_fiss_2023.gpkg
+form_fiss_site_raw <- fpr::fpr_sp_gpkg_backup(
+  path_gpkg = paste0("~/Projects/gis/", dir_project, '/data_field/2023/form_fiss_site_2023.gpkg'),
+  update_utm = TRUE,
+  update_site_id = FALSE,
+  write_back_to_path = FALSE,
+  write_to_csv = FALSE,
+  write_to_rdata = FALSE,
+  return_object = TRUE)
+
+
+## Import the habitat_confirmations.xlsm
+habitat_confirmations <- fpr::fpr_import_hab_con(path = '~/Projects/repo/fish_passage_skeena_2023_reporting/data/habitat_confirmations.xls', row_empty_remove = T)
+```
+
+## Extract temp, cond, and turb data
+
+We need to extract temp, cond, turb data from `step_2_fish_coll_data` sheet in the `habitat_confirmations.xls` file. 
+
+As per [PR feedback](https://github.com/NewGraphEnvironment/fish_passage_skeena_2023_reporting/pull/68#discussion_r1583971038), I used `all_of()` instead of `c()` to select columns, but turns out you still need to use `c()` within `all_of()` to select columns if the column names are not stored in variables or in a vector, so I made a vector of column names. Not sure if this is worth it because we only use this vector once, but may be nice to reproducibility in the future.
```

Comment:
Worth it due to the safety it provides! Making a vector with `c` and calling a list of columns to be selected using `all_of` vs `c` are two very separate things. It's worth mentioning the safety thing here (next time) if you are bringing it up since it is the reason for the comment on the PR in the first place.


You should also be aware of `dplyr::across` and `dplyr::any_of`. You can see examples of when we use terms like `across`, `left_join` etc using `Find in Files` from the `Rtudio/edit` menu.

![image](https://github.com/NewGraphEnvironment/fish_passage_skeena_2023_reporting/assets/46538103/cc0e1c35-b75c-4324-9b88-e20949de8a17)

Reply:
okay sweet I will add a note about why we use `all_of` vs `c`

Reply:
no need on thiis one though. we are good